.github/workflows/actionlint.yaml

```
@@ -0,0 +1,33 @@
+name: Actionlint
+
+on:
+  pull_request:
+    paths:
+      - ".github/workflows/**"
+  push:
+    branches: main
+    paths:
+      - ".github/workflows/**"
+  workflow_dispatch:
+
+# Restrict permissions for GITHUB_TOKEN to minimum required
+permissions:
+  contents: read
+  # All other permissions are implicitly none
+
+jobs:
+  actionlint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false # Because installing actionlint runs arbitrary code, we discard the GITHUB_TOKEN after this step
+      # Taken from https://github.com/rhysd/actionlint/blob/v1.7.7/docs/usage.md#use-actionlint-on-github-actions
+      - name: Download actionlint
+        id: get_actionlint
+        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
+        shell: bash
+      - name: Check workflow files
+        run: ${{ steps.get_actionlint.outputs.executable }}
+        shell: bash
```

```suggestion
      - name: Check workflow files
        run: nix-shell -p actionlint --run actionlint

```

I wonder whether we can just do something like this? Ideally using the same version of Nixpkgs as for anything else to avoid another download.

---

Ah, we need the nix-install-action here first...

---

btw here's a cumbersome but workable way to sync the nixpkgs version: https://github.com/ngi-nix/ngipkgs/blob/52c0b887994e3f497ef21878ccfc2c7b8b292d5c/.github/workflows/test-demo.sh#L34-L39

---

@fricklerhandwerk> sorry I'm not Nix-savvy enough to fully adapt the snippet to this repo. Can you correct the following:

```shell
      - name: Check workflow files
        run: |
          nixpkgs_revision=$(nix-instantiate --eval --attr sources.nixpkgs.rev /ngipkgs | jq --raw-output)
          NIXPKGS="https://github.com/NixOS/nixpkgs/archive/$nixpkgs_revision.tar.gz"
          nix-shell --include nixpkgs="$NIXPKGS" --packages nix --run "actionlint"
```

Can you adapt the `nixpkgs_revision` definition to this repo :grimacing:?

---

@fricklerhandwerk> sorry this is beyond me, I have no idea what `nix-instantiate` does for example. So either we should drop this PR, or ask someone more savvy to tackle this thread :slightly_smiling_face:

---

No problem, I'll pick it up then. It's low priority, so no timeline.